search disabled users by display name or email

hweihwang opened the pull request:
https://github.com/nextcloud/user_saml/issues/994

blizzz commented on lib/GroupBackend.php:
```diff
@@ -7,6 +7,7 @@
 
 namespace OCA\User_SAML;
 
+use OC\User\LazyUser;
```
This is problematic, it is not part of the API. It is also bad that  the internal backend makes use of it, while there is no chance for external ones. 

Would it be sufficient to implement `IProvideEnabledStateBackend` instead? Then the user management abstraction would create the LazyUser objects.

blizzz commented on lib/GroupBackend.php:
```diff
@@ -154,15 +154,35 @@ public function groupExistsWithDifferentGid(string $samlGid): ?string {
+				)
+			);
+
+			$searchParam1 = $query->createNamedParameter('%' . $this->dbc->escapeLikeParameter($search) . '%');
+			$searchParam2 = $query->createNamedParameter('%' . $this->dbc->escapeLikeParameter($search) . '%');
+			$searchParam3 = $query->createNamedParameter('%' . $this->dbc->escapeLikeParameter($search) . '%');
+			$query->andWhere(
+				$query->expr()->orX(
+					$query->expr()->ilike('m.uid', $searchParam1),
+					$query->expr()->ilike('dn.value', $searchParam2),
+					$query->expr()->ilike('em.value', $searchParam3)
```
having one $searchParam should be sufficient, as they are all the same.

CarlSchwan replied:
iirc this does not work with either oracle or PostgreSQL, I had unit tests fail in the past on one of them because I tried to be clever xD

CarlSchwan commented:
Need a rebase ;)

hweihwang committed:
fix: search disabled users by display name or email

Extend user and group search to match display names and email addresses, keep the enabled-state backend integration from master, and add regression coverage for the new lookup paths. Also skip the unsupported OCI 11 + Nextcloud master matrix entry because server master now sets Oracle sessions to Etc/UTC, which OCI 11 cannot resolve.

Signed-off-by: Hoang Pham <hoangmaths96@gmail.com>

blizzz commented:
@hweihwang does it need a backport to stable-7 for 31 or 30?
